File: src/main/java/com/deneb/client/command/commands/Friend.java
package com.deneb.client.command.commands;

import com.deneb.client.client.FriendManager;
import com.deneb.client.command.Command;
import com.deneb.client.utils.ChatUtil;
import com.deneb.client.utils.EntityUtil;
import com.deneb.client.utils.Wrapper;
import net.minecraft.entity.player.EntityPlayer;

/**
 * Created by killRED on 2020
 * Updated by B_312 on 01/15/21
 */
@Command.Info(command = "friend", description = "Friend command.")
public class Friend extends Command {

    @Override
    public void onCall(String s, String[] args) {
        try {
            if (args[0].equalsIgnoreCase("all")) {
                for (Object object : EntityUtil.getEntityList()) {
                    if (object instanceof EntityPlayer) {
                        EntityPlayer player = (EntityPlayer) object;
                        if (EntityUtil.isFakeLocalPlayer(player)) {
                            continue;
                        }
                        if (!player.isInvisible()) {
                            FriendManager.addFriend(EntityUtil.getPlayerName(player));
                        }
                    }
                }
            } else if (args[0].equalsIgnoreCase("get")) {
                ChatUtil.sendNoSpamMessage(FriendManager.getFriendStringList().toString());
            } else if (args[0].equalsIgnoreCase("add")) {
                boolean addedFriend = false;
                for (EntityPlayer player : Wrapper.getMinecraft().world.playerEntities) {
                    if (!player.getName().toLowerCase().equals(args[1].toLowerCase())) {
                        continue;
                    }
                    FriendManager.addFriend(player.getName());
                    addedFriend = true;
                }
                if (!addedFriend) {
                    ChatUtil.sendNoSpamErrorMessage("Player " + ChatUtil.SECTIONSIGN + "b" + args[1] + ChatUtil.SECTIONSIGN + "r" + " does not exist");
                }
            } else if (args[0].equalsIgnoreCase("remove")) {
                FriendManager.removeFriend(args[1]);
            } else {
                ChatUtil.sendNoSpamErrorMessage(getSyntax());
            }

        } catch (Exception e) {
            ChatUtil.sendNoSpamErrorMessage(getSyntax());
        }
    }

    @Override
    public String getSyntax() {
        return "friend <add/all/get/remove>";
    }

}

File: src/main/java/club/deneb/client/command/commands/Friend.kt
package club.deneb.client.command.commands

import club.deneb.client.client.FriendManager.addFriend
import club.deneb.client.client.FriendManager.friendList
import club.deneb.client.client.FriendManager.removeFriend
import club.deneb.client.command.Command
import club.deneb.client.utils.ChatUtil
import club.deneb.client.utils.ChatUtil.sendNoSpamErrorMessage
import club.deneb.client.utils.ChatUtil.sendNoSpamMessage
import club.deneb.client.utils.EntityUtil
import net.minecraft.client.Minecraft
import net.minecraft.entity.player.EntityPlayer

/**
 * Created by killRED on 2020
 * Updated by B_312 on 01/15/21
 */
@Command.Info(command = "friend", description = "Friend command.")
class Friend : Command() {
    override fun onCall(s: String, vararg args: String) {
        try {
            if (args[0].equals("all", ignoreCase = true)) {
                for (objects in EntityUtil.getEntityList()) {
                    if (objects is EntityPlayer) {
                        if (EntityUtil.isFakeLocalPlayer(objects)) {
                            continue
                        }
                        if (!objects.isInvisible) {
                            addFriend(EntityUtil.getPlayerName(objects))
                        }
                    }
                }
            } else if (args[0].equals("get", ignoreCase = true)) {
                sendNoSpamMessage(friendList().toString())
            } else if (args[0].equals("add", ignoreCase = true)) {
                var addedFriend = false
                for (player in Minecraft.getMinecraft().world.playerEntities) {
                    if (!player.name.equals(args[1], ignoreCase = true)) {
                        continue
                    }
                    addFriend(player.name)
                    addedFriend = true
                }
                if (!addedFriend) {
                    sendNoSpamErrorMessage("Player " + ChatUtil.SECTIONSIGN + "b" + args[1] + ChatUtil.SECTIONSIGN + "r" + " does not exist")
                }
            } else if (args[0].equals("remove", ignoreCase = true)) {
                removeFriend(args[1])
            } else {
                sendNoSpamErrorMessage(getSyntax())
            }
        } catch (e: Exception) {
            sendNoSpamErrorMessage(getSyntax())
        }
    }

    override fun getSyntax(): String {
        return "friend <add/all/get/remove>"
    }
}